crates/bevy_animation/src/animation_curves.rs

```diff
@@ -199,7 +199,7 @@ pub trait AnimatableProperty: Send + Sync + 'static {
 
     /// The [`EvaluatorId`] used to look up the [`AnimationCurveEvaluator`] for this [`AnimatableProperty`].
     /// For a given animated property, this ID should always be the same to allow things like animation blending to occur.
-    fn evaluator_id(&self) -> EvaluatorId;
+    fn evaluator_id(&self) -> EvaluatorId<'_>;
```

Comment:
This is a very standard case: we're secretly returning an object with a lifetime bound to self.

Good lint!

Reply:
This is nice. It always bugged me that you couldn't tell the difference between a thing with a lifetime tied to self and something that was static.

Reply:
This is great. Pretty much sells me on this lint.